utils/documentOps.py

```diff
@@ -142,6 +174,26 @@ def print_row(array):
         str += a + " |"
     print(str)
 
+# Min opset: if explicit ==MIN== value, use as is. Otherwise min of default and
+# custom versions delivered in opset_dict.
+def compute_min_opset(op):
+    if op in min_dict:
+        return min_dict[op]
+    min_list = [min_opset_default]
+    if op in opset_dict:
+        min_list.extend(opset_dict[op])
+    return min(min_list)
+
+# Max opset: if explicit ==MAX== value, use as is. Otherwise max of default and
+# custom versions delivered in opset_dict.
+def compute_max_opset(op):
+    if op in max_dict:
+        return max_dict[op]
+    max_list = [max_opset_default]
```

Comment:
Instead of setting a hardcoded value for the max_opest... why not use the largest or first element of that operation in the `version_dict` which is located in  onnx-mlir/utils/gen_onnx_mlir.py. This value should be accurate and would essentially reflect what is currently being supported. Not every op has support for 18, so I do not think this should be hardcoded in.

Reply:
However, if you think this would be too much to do the above... I would prefer using NA as the default values then hardcoding values we can not be sure of. NA is a given standard and we can easily make the updates once min/max are explicitly defined.

Reply:
@hamptonm1, the value in `version_dict` only say "we now support an op that is incompatible with prior ops"... Most op that get updated from one onset to the next do not break compatibility, and thus don't show up in the version_dict even though we did quite a bit of work to make them work.

My suggestion: I would put the default min to a set that we largely support for most ops, and default max to the current onset (say 19). And then whenever there is an op that we don't support in the latest, we say MAX=13 (if we stop supporting at 14 and higher). Then if we fix this op to support 19, we simply remove the MAX. 

If you prefer to rename "MAX 13" to "NO_SUPPORT 14" (as last supported of an op at 13 means that we don't support 14...) that is fine by me too.

Reply:
> @hamptonm1, the value in `version_dict` only say "we now support an op that is incompatible with prior ops"... Most op that get updated from one onset to the next do not break compatibility, and thus don't show up in the version_dict even though we did quite a bit of work to make them work. If it is confusing, I would suggest to remove looking at these values and simply rely on the default min/max and the explicit min/max.
> 
> My suggestion: I would put the default min to a set that we largely support for most ops, and default max to the current onset (say 19). And then whenever there is an op that we don't support in the latest, we say MAX=13 (if we stop supporting at 14 and higher). Then if we fix this op to support 19, we simply remove the MAX.
> 
> If you prefer to rename "MAX 13" to "NO_SUPPORT 14" (as last supported of an op at 13 means that we don't support 14...) that is fine by me too.

This is what is commented in the script above `version_dict`

```
# Record the version of each operation that is treated as the current version.
# To check whether the onnx package being used has newer version operation,
# run this script with --check-operation-version flag.
# Update this dictionary when a newer version is implemented
# TODO: how to keep the old version
```

I am assuming the exercise for this is to specify the minimum version we do support and the max version we support which varies based on operation. So it does not make sense to hardcode a value that may not be applicable to said operation. That I am not understanding. 

From your example of MAX.... I would be very clear in terms of what is supported here. So let's say minimum version support for MAX is 6 and the max is 13 as you mentioned 14 is not supported. The MIN should be 1 and MAX 13. This makes more sense to me than having "NO_SUPPORT_14".Min and max should be the inclusive range we support. So the lowest value up until the highest value. Just because we have Opset 19 available does not mean every operation is compatible with version 19. So this would create more confusion than understanding if we were to hard code values.